src/qtcore/qml/qml.js

```diff
@@ -321,6 +321,30 @@ function setupGetterSetter(obj, propName, getter, setter) {
   });
 }
 
+function connectSignal(item, signalName, value, objectScope, componentScope) {
+    if (!item[signalName]) {
+        console.warn("No signal called " + signalName + " found!");
+        return;
+    }
+    else if (typeof item[signalName].connect != 'function') {
+        console.warn(signalName + " is not a signal!");
+        return;
+    }
+    if (!value.eval) {
+        var params = "";
+        for (var j in item[signalName].parameters) {
+            params += j==0 ? "" : ", ";
+            params += item[signalName].parameters[j].name;
+        }
+        value.src = "(function(" + params + ") { _executionContext = __executionContext;" + value.src + "})";
+        value.isFunction = false;
+        value.compile();
+    }
+    var f = value.eval(objectScope, componentScope);
```

Comment:
Could you give the variablie `f` a bit more semantic name here? It's not clear what it is.